pkl-core/src/main/java/org/pkl/core/http/HttpClient.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.http;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpTimeoutException;
+import java.nio.file.Path;
+import javax.net.ssl.SSLContext;
+
+/**
+ * An HTTP client.
+ *
+ * <p>To create a new HTTP client, use a {@linkplain #builder() builder}. To send {@linkplain
+ * HttpRequest requests} and retrieve their {@linkplain HttpResponse responses}, use {@link #send}.
+ * To release resources held by the client in a timely manner, use {@link #close}.
```

Comment:
> I'm proposing that both here, and in evaluator builder, we provide knobs for addCertificates, and possibly any other options on HttpClient.Builder.

There's already a `CliBaseOptions.caCertificates` constructor parameter. There is currently no builder for `CliBaseOptions`.
Adding convenience methods to `EvaluatorBuilder` is possible, but let's defer this. I could do this in #197 , which already adds a couple of convenience methods to `EvaluatorBuilder`.

> the wording is slightly vague

It's because the underlying JDK client only offers weak guarantees, and at least connections will eventually expire even if `close()` isn't called. (Before JDK 21 there isn't even a `close()` method.) But I'll remove this nevertheless.